File: frontend/market/src/app/model/mobile/mobile-dto.ts
import {MobileMainImage, MobileNotMainImage} from "./mobile-image";

export class MobileDto {
  constructor(public mobileId?: string,
              public brand?: string,
              public model?: string,
              public os?: string,
              public screenSize?: string,
              public displayResolution?: string,
              public displayTechnology?: string,
              public ram?: number,
              public storageCapacity?: number,
              public chipsetModel?: string,
              public cameraResolution?: string,
              public simCardSlot?: string,
              public battery?: number,
              public color?: string,
              public releaseYear?: string,
              public mainImage?: MobileMainImage) {}
}

export class MobileDtoFull {
  constructor(public mobileId?: string,
              public brand?: string,
              public model?: string,
              public os?: string,
              public screenSize?: string,
              public displayResolution?: string,
              public displayTechnology?: string,
              public ram?: number,
              public storageCapacity?: number,
              public chipsetModel?: string,
              public cameraResolution?: string,
              public simCardSlot?: string,
              public battery?: number,
              public color?: string,
              public releaseYear?: string,
              public type?: string,
              public osVersion?: string,
              public processorClockFrequency?: number,
              public coresNumber?: number,
              public technicalProcess?: number,
              public housingMaterial?: string,
              public simFormat?: string,
              public length?: number,
              public width?: number,
              public height?: number,
              public weight?: number,
              public mainCamerasNumber?: number,
              public builtInFlash?: boolean,
              public automaticFocus?: boolean,
              public opticalStabilization?: boolean,
              public mainCamera?: boolean,
              public mainCameraAperture?: string,
              public frontCamera?: boolean,
              public frontCameraResolution?: string,
              public frontCameraAperture?: string,
              public gps?: boolean,
              public glonass?: boolean,
              public beidou?: boolean,
              public edge?: boolean,
              public hspa?: boolean,
              public hspaPlus?: boolean,
              public lte?: boolean,
              public fiveG?: boolean,
              public bluetooth?: boolean,
              public bluetoothVersion?: string,
              public audioOutput?: boolean,
              public audioOutputVersion?: string,
              public wifi?: boolean,
              public wifiVersion?: string,
              public connection?: string,
              public nfc?: boolean,
              public batteryType?: string,
              public chargeTime?: string,
              public mainImage?: MobileMainImage,
              public notMainImages?: Array<MobileNotMainImage>
              ) {}
}

File: backend/model/src/main/java/com/beacon/model/dtos/MobileDto.java
package com.beacon.model.dtos;

import com.beacon.model.MobileMainImage;
import lombok.Getter;
import lombok.NoArgsConstructor;

/**
 * It is a dto of {@code Mobile}. The dto includes all fields of {@code Mobile} and count of offers and minimal price.
 */
@Getter
@NoArgsConstructor
public class MobileDto {

    private String mobileId;

    private String brand;

    private String model;

    private String os;

    private String screenSize;

    private String displayResolution;

    private String displayTechnology;

    private Integer ram;

    private Integer storageCapacity;

    private String chipsetModel;

    private String cameraResolution;

    private String simCardSlot;

    private Integer battery;

    private String color;

    private String releaseYear;

    private Integer relevant;

    private MobileMainImage mainImage;

    private Long countOfOffers;

    private Double minimalPrice;

    public MobileDto(String mobileId, String brand, String model, String os, String screenSize, String displayResolution,
                     String displayTechnology, Integer ram, Integer storageCapacity, String chipsetModel,
                     String cameraResolution, String simCardSlot, Integer battery, String color, String releaseYear,
                     Integer relevant, Long imageId, byte[] image, Long countOfOffers, Double minimalPrice) {
        this.mobileId = mobileId;
        this.brand = brand;
        this.model = model;
        this.os = os;
        this.screenSize = screenSize;
        this.displayResolution = displayResolution;
        this.displayTechnology = displayTechnology;
        this.ram = ram;
        this.storageCapacity = storageCapacity;
        this.chipsetModel = chipsetModel;
        this.cameraResolution = cameraResolution;
        this.simCardSlot = simCardSlot;
        this.battery = battery;
        this.color = color;
        this.releaseYear = releaseYear;
        this.relevant = relevant;
        this.countOfOffers = countOfOffers;
        this.minimalPrice = minimalPrice;
        this.mainImage = new MobileMainImage(imageId, image);
    }
}
